dev/plugins/hu.elte.txtuml.diagnostics/src/hu/elte/txtuml/diagnostics/animation/js/DiagnosticsServer.java

```diff
@@ -0,0 +1,127 @@
+package hu.elte.txtuml.diagnostics.animation.js;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import java.util.Map.Entry;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.stream.Collectors;
+
+import com.sun.net.httpserver.Headers;
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
+
+import hu.elte.txtuml.api.model.execution.diagnostics.protocol.GlobalSettings;
+import hu.elte.txtuml.api.model.execution.diagnostics.protocol.MessageType;
+import hu.elte.txtuml.api.model.execution.diagnostics.protocol.ModelEvent;
+import hu.elte.txtuml.diagnostics.session.DiagnosticsPlugin;
+
+/**
+ * Serves diagnostics data over HTTP.
+ */
+public class DiagnosticsServer {
+
+	private HttpServer server;
+	private ConcurrentMap<String, RegistryEntry> registry = new ConcurrentHashMap<>();
+	private DiagnosticsPlugin diagnosticsPlugin;
+	
+	public DiagnosticsServer(DiagnosticsPlugin diagnosticsPlugin){
+		this.diagnosticsPlugin = diagnosticsPlugin;
+	}
+
+	public void start(Integer port) throws IOException {
+		server = HttpServer.create(new InetSocketAddress(port), 0);
+		server.createContext("/" + GlobalSettings.TXTUML_DIAGNOSTICS_HTTP_PATH, new DiagnosticsHandler());
+		server.createContext("/delay", new DelayHandler());
```

Comment:
Please introduce _"delay"_ as a final field in `GlobalSettings`, see one row above.